Makefile

```
@@ -25,6 +25,16 @@ json-schema.tar.gz: $(OUT)
 	tar -czf json-schema.tar.gz --exclude '.*' json-schema
 	rm -rf json-schema
 
+venv: $(VENV)
+
+$(VENV):
+	python -m venv .venv
+	. .venv/bin/activate && python -m pip install --upgrade pip setuptools wheel
+	. .venv/bin/activate && python -m pip install -r requirements.txt
```

No need to use the activation script in a non-interactive mode. You can just run things from virtualenvs directly.
```suggestion
	.venv/bin/python -m pip install -r requirements.txt
```

---

Related to my previous comment: `$@/bin/python -m pip install -r requirements.txt` is even better